--- vendor/github.com/gogo/protobuf/types/zz_generated.deepcopy.orig.go	2019-06-14 11:51:52.000000000 -0700
+++ vendor/github.com/gogo/protobuf/types/zz_generated.deepcopy.go	2019-06-14 11:58:32.204535994 -0700
@@ -750,7 +750,7 @@
 func (in *Value) DeepCopyInto(out *Value) {
 	*out = *in
 	if in.Kind != nil {
-		out.Kind = in.Kind.DeepCopyisValue_Kind()
+		out.Kind = in.Kind
 	}
 	out.XXX_NoUnkeyedLiteral = in.XXX_NoUnkeyedLiteral
 	if in.XXX_unrecognized != nil {
